### PR TITLE
reproject all layers to aggregation CRS

### PR DESCRIPTION
**Pull request by Gustry**
### What does it fix?
* Ticket: part of #4437 
* Funded by: DFAT
* Description: We decided to reproject all layers to aggregation CRS or custom CRS if no aggregation provided.

I'm not sure why I needed to update the test `test_aggregate_post_processors_vector`. We lost 100 people in the analysis. But this is related to some small side effects of reprojection. For me, it's OK.

Ready to review

### Checklist:
- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Add to the changelog in metadata.txt if it's a new feature
- [ ] Unit test for new code added
- [ ] Request someone to review or test your PR

### CODE REVIEW

File: safe/report/test/test_impact_report.py [hunk 1]
```diff
@@ -54,7 +54,7 @@
 
 QGIS_APP, CANVAS, IFACE, PARENT = get_qgis_app()
 
-from qgis.core import QgsMapLayerRegistry
+from qgis.core import QgsMapLayerRegistry, QgsCoordinateReferenceSystem
```
Comment (ismailsunni) [inline safe/report/test/test_impact_report.py]: __[flake8]__
* [E402] module level import not at top of file

File: safe/gis/raster/test/test_zonal_statistics.py [hunk 2]
```diff
@@ -8,7 +8,8 @@
 )
 QGIS_APP, CANVAS, IFACE, PARENT = get_qgis_app()
 
-from qgis.core import QGis
+from qgis.core import QGis, QgsCoordinateReferenceSystem
```
Comment (ismailsunni) [inline safe/gis/raster/test/test_zonal_statistics.py]: __[flake8]__
* [E402] module level import not at top of file

File: safe/gis/raster/test/test_zonal_statistics.py [hunk 3]
```diff
@@ -8,7 +8,8 @@
...
+from safe.gis.vector.reproject import reproject
```
Comment (ismailsunni) [inline safe/gis/raster/test/test_zonal_statistics.py]: __[flake8]__
* [E402] module level import not at top of file

**Comment by Gustry** [comment]
Tests from @ismailsunni are now failing here ... ahah
```
AssertionError: <safe.impact_function.impact_function.ImpactFunction object at 0x7f65348d4310> != <safe.impact_function.impact_function.ImpactFunction object at 0x7f65348d4310>
```

It's the same address. It looks suspicious, no? How two objects sharing the same address, ( so same address in memory can be different) I can have a look.

**Comment by ismailsunni** [comment]
> It's the same address. It looks suspicious, no? How two objects sharing the same address, ( so same address in memory can be different) I can have a look.

Yeah, it's strange. Let me try locally here also.

**Comment by ismailsunni** [comment]
Hi @Gustry 
The log message is 
```
2017-10-18 05:35:36,167 - InaSAFE - ERROR - Property requested_extent_crs is not found. The exception is 'ImpactFunction' object has no attribute 'requested_extent_crs'
```

It seems you remove the `requested_extend_crs` from the impact function, but I use it as a property to check.

If you change `requested_extend_crs` to `crs` in the `def __eq__` you will make the unit test pass.

**Comment by Gustry** [comment]
Sorry yes, obvious now. And I noticed that change in your PR, that's why I said I wanted to rebase this PR to see what could happen.

But as I can't run the test in pycharm, I didn't see the log. Thanks

**Comment by ismailsunni** [comment]
Hi @Gustry 
For the failed unit test, I tried in my local. Although it's quite strange (only  happenened when I run from `test_scenario_directory` but not from `test_scenarios`. The scenario is `polygon_continuous_on_line.json`

Since I know that you couldn't run unit test, and I think I know how to fix the issue, I make a commit here to fix the issue. Basically, it's my bad that I forget to put the crs for deserialization IF. Here is my commit to fix it : https://github.com/ismailsunni/inasafe/commit/4271ca97efde70cfde1c981703b95b412fb55e16

**Comment by Gustry** [comment]
Still not happy, and something missing with docker, it's to be able to run test with the debugger.
Feel free to push to my branch if you want.

I really need to fix my pycharm.

**Pull request merged**